### DISCUSSION TITLE
Welcome to Discussions of DGUS-reloaded(ForKlipper)_CR6Community Edition

### DISCUSSION THREAD

**Discussion by Thinkersbluff** [Thinkersbluff/DGUS-Reloaded_for_CR6-DWIN-SET_Component#3] [Announcements]
# Welcome!

This repo is one of two, at the moment, which jointly host the two parts of the CR6Community Edition of DGUS-Reloaded.

The goal of this project is to restore some functionality to the stock DWIN screen, after converting your CR6-series 3D printer to run on Klipper.  

The first versions of this DWIN_SET UI were achieved by refactoring the landscape mode T5UID1 DGUS-Reloaded DWIN App by Desuuuu.  This allowed me to learn the architecture of the two parts and confirm that I would be able to generate a CR6-compatible version with a reasonable amount of functionality.

At version 0.3 and subs, the look and feel of this UI is inspired by [the Marlin-based CR6 Community Firmware [UI]](https://github.com/CR6Community/CR-6-touchscreen).   Some Klipper functionality is different from Marlin, so the UI will not be 100% identical, but ;the workflow and user-experience concepts baked into the CF6.1 UI are very evident in this UI, as of v0.3.  
The deliberate alignment/overlap between the two UI projects is intended to keep the learning curve shallow for CR6 Community members who decide to switch from CF6.1 to Klipper on their CR6 printer(s).

The Klipper portion of this solution is a fork of a fork of the Klipper3D master. 

- The first fork is ["DGUS-Reloaded-Klipper" , by [Desuuuu]](https://github.com/Desuuuu/DGUS-reloaded-Klipper).
- This 2nd fork is made necessary by the architecture of the t5uid1/dgus-reloaded application and by the need to compile a modified Klipper.bin to support these communications.

Hopefully, the long-term version of the CR6 Community Edition will NOT always depend upon maintaining a unique fork of Klipper.  
At this time, however, I do not know of any viable alternative approach.   Perhaps you, dear Community Member, have some specialized knowledge/experience which enables you to be of some assistance, in that regard?  If so, PLEASE start a Discussion on that topic & let's get ourselves back onto the mainstream Klipper repo.

-->

**Comment by xXHexaKillXx** [comment]
oh wow can i have a few questions. since this repo is another repo from dgus. Is it up to date with the klipper firmware?

**Reply by Thinkersbluff** [reply]
Sorry, I did not see your question earlier...   Yes, the firmware continues to work with the latest Klipper.  I have since also added documentation explaining how to "pin" your Klipper to the last known compatible version until I release a fix, any time that Klipper3D disable or refactor Klipper features on which the firmware depends.

**Comment by Thinkersbluff** [comment]
The DWIN component and Klipper component each reside in their own repo, for now.  Your question is answered in the Klipper repo Readme.

Now that we have uploaded the precompiled Klipper.bin files, there is no need to keep updating the modified Klipper.  It is best to let Moonraker keep Klipper up to date, and just copy t5uid1 into klippy/extras.

**Reply by xXHexaKillXx** [reply]
i am just asking because i was having trouble with my dwin. i was trying to use fluidd with dwin enabled with dgus. when dgus was enabled fluidd just refused to connect. am i doing something wrong?

**Reply by xXHexaKillXx** [reply]
the printer is basically a clone creality cr10 of a clone of alfwise u30 pro. known as longer lk 4 pro

**Reply by Thinkersbluff** [reply]
I don’t use Fluidd, so I don’t know whether there is an issue with it.
I uploaded Mansail configuration files in the Related Changes folder.  I moved the [t5uid1] section of printer.cfg into its own dgus-reloaded.cfg file to facilitate commenting-out that include instruction, until Klipper is working correctly.

I install and configure Klipper per their documentation, first.  Then I activate t5uid1 by uncommenting that include statement.

**Reply by Thinkersbluff** [reply]
I am only building and maintaining this version specifically for CR6 printers.  There are too many printer models out there for me to try to support them all.  
You may be able to build your own Klipper.bin file from the Klipper component, but unless your screen is the same as the cr6 one, you will need to refactor and maintain your own variant of this DWIN component.

**Reply by xXHexaKillXx** [reply]
is there any additional stuff i need to do to be able to move dgus code that use to be in the config file. and where do you save it to be called from config?

**Reply by xXHexaKillXx** [reply]
how do you flash the MCU since the uart is not present?

**Reply by xXHexaKillXx** [reply]
actually don't reply i wasn't looking hard enough you have the other repo project.

**Reply by Thinkersbluff** [reply]
You need to follow the Klipper3D.org instructions or Fluidd instructions for basic Klipper/Fluidd questions.  
e.g. you flash the mcu from an sd card, when you are not able to flash via usb.

On the Klipper host, most of the config files are in ~/printer_data/config.

**Reply by xXHexaKillXx** [reply]
the method actually really helpful. i was fortunate enough to stumble your github. if i did not i would end up getting a touch screen for my pi instead and basically throw away a perfectly still good working screen

**Reply by xXHexaKillXx** [reply]
am i able to see your example of your config file?

**Reply by xXHexaKillXx** [reply]
how do you compile the .bin if the dgus reload is not to be seen during the compile uart part? do you compile with dgus first then update the klipper?

**Reply by Thinkersbluff** [reply]
Not sure whether I correctly understand your question about compilation. 

I used KIAUH to install my modified Klipper onto the pi host (that is why I provide a repos.txt file for KIAUH) Then I used Make Menuconfig and Make, to build my own klipper.bin file, which I flashed to my printer mcu, using an sd card. Then I reverted KIAUH back to using the Klipper3D.org/master repo, and installed the current Klipper, which also works with that modified klipper.bin. (Note:  it took me a couple of tries, to find the correct settings for the DWIN serial port, in Make Menuconfig.)

Since KIAUH erases the previous Klipper directory before installing a new one, I also had to reinstall the klippy/extras after each installation of Klipper. 

Is that what you were asking?

**Reply by Thinkersbluff** [reply]
> am i able to see your example of your config file?

They are in the Klipper component Repo, Related Changes sub folder. Pick a motherboard sub folder then look in “Overwrite these Klipper host files”, then open the sub folder whose title includes “Use these files to tailor…”

**Reply by xXHexaKillXx** [reply]
alright i think i just got it running thanks. your file also works for AVR boards as confirmed @Thinkersbluff

**Reply by xXHexaKillXx** [reply]
mega2560 board is confirm to be working.

**Reply by xXHexaKillXx** [reply]
I am gonna try and tweak the config file first and see if fluidd also runs. Will report more to come. I suggest rewording your document. It took me awhile to under stand. From a user that is unexperience it really is not plug and play and a lot of asking as i for mention. Let me re-word your doc installation so it is nore clear for a unexperience person can read.

**Reply by Thinkersbluff** [reply]
> ..... I suggest rewording your document. It took me awhile to under stand. From a user that is unexperience it really is not plug and play and a lot of asking as i for mention. Let me re-word your doc installation so it is nore clear for a unexperience person can read.

That installation document is definitely a work in progress.  All help gratefully received!

Glad you were able to make one of the klipper.bin files work for you.

For the sake of others who find this later:
- your printer is a Longer lk 4 pro
- the motherboard processor is a mega2560, yet one of the stm32f processor klipper.bin files is working for you.

Is that the Creality klipper.bin file?
What part number is your motherboard? LGT_KIT_V1.0?

I see that board also has TMC2208 drivers, not the TMC2209 on the CR6 boards.

Interesting that one of the CR6 bin files is working for you.

**Reply by xXHexaKillXx** [reply]
The board is indeed a lgt kit v1.0 but the motor being used is different slightly. I had to compile my own fimware. Since the board uses meg2560 it spat out as .hex file instead.

**Reply by xXHexaKillXx** [reply]
I will do my write up progress for you. Feel free to use it to your liking. I just add stuff too simplify your steps and support for fluidd as well. For now i got mainsail working as intended. I was gettin frustrated at the start. Went to bed right after and woke up and found out what i was doing wrong. Bam i got it working. But config file not completely finished tho, i still need to configure the step revolution for step motor.

**Reply by Thinkersbluff** [reply]
> The board is indeed a lgt kit v1.0 but the motor being used is different slightly. I had to compile my own fimware. Since the board uses meg2560 it spat out as .hex file instead.

I am still confused, then.
Did you create your own klipper.bin file, using the modified Klipper and make menuconfig?

**Reply by xXHexaKillXx** [reply]
Yes. I will type a written doc how i did mine you will find it much easier to understand
